samples/boards/nordic/system_off/src/main.c

```diff
@@ -87,6 +87,19 @@ int main(void)
 		retained_update();
 	}
 
+	/* Wait until all logs are printed into console */
+	if (IS_ENABLED(CONFIG_PM_DEVICE) && IS_ENABLED(CONFIG_SERIAL)) {
+		static const struct device *dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_console));
+		int err;
+		enum pm_device_state state;
+
+		if (dev) {
+			do {
+				err = pm_device_state_get(dev, &state);
+			} while ((err == 0) && (state == PM_DEVICE_STATE_ACTIVE));
+		}
+	}
```

Comment:
I don't quite understand this piece of code. Why is the console device supposed to change its PM state by itself here?
And actually, shouldn't it be already suspended after https://github.com/zephyrproject-rtos/zephyr/blob/b0503baf967854ce8e35de9a0548490e7d16ca6c/samples/boards/nordic/system_off/src/main.c#L78-L82?

Reply:
I just removed that commit.
Discussed with Krzysztof, it was not even active - PM flag is not set.
I was seeing some delayed logs when using configuration with automatic wake ups (GRTC) but this could be my error when observing incoming logs.